pubsub/batcher/batcher.go

```diff
@@ -283,5 +298,13 @@ func (b *Batcher) Shutdown() {
 	b.mu.Lock()
 	b.shutdown = true
 	b.mu.Unlock()
+
+	// On shutdown, ensure that we attempt to flush any pending items
+	// if there's a minimum batch size.
+	if atomic.LoadInt32(&b.nHandlers) < int32(b.opts.MaxHandlers) {
```

Comment:
What happens if nHandlers == MaxHandlers? Won't we drop some messages then?

Reply:
At this point, `nextBatch` in the handlers call will return the remaining items as `shutdown` is set to true, so everything will be processed as expected.

Reply:
If nHandlers == MaxHandlers, `nextBatch` won't even be called...?

Reply:
It's not obvious to me that we should be checking `nHandlers` here at all. Seems like during shutdown we need to choose between possibly creating more than MaxHandlers handlers, or dropping messages.